Docs number text date component (#1892)
Altinn/altinn-studio-docs: content/altinn-studio/reference/ux/components/Date/_index.en.md

---
title: Date
linktitle: Date
description: A component that allows you to display formatted date with or without label.
schemaname: Date # Component schema name used to autogenerate list of properties from json schema (replace with appropriate component name)
weight: 10 # Do not change, the components will be sorted alphabetically
toc: true
---

## Usage

> The Date component allows users to add structured formatted date, with or without label

Reviewer: ```suggestion
The `Date` component allows users to add structured formatted date, with or without label.
```

| `value`                      | string   | The date you want to display. Must be a ISO6801 string.                                  |
| `textResourceBindings.title` | string   | Label of the date you want to display                                                    |
| `format`                     | string   | A formatting string based on [Unicode Technical standard](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table)                              |
| `direction`                  | string   | Sets the structured direction of label and value. **Enum:** `"horizontal" \| "vertical"` |
| `icon`                       | string   | An URL path to the image/icon                                                            |

## Configuration

### Add component

```json{hl_lines="6-"}
{
  "id": "date",
  "type": "Date",
  "textResourceBindings": {
    "title": "Date"
  },
  "value": "2022-09-27 18:00:00.000",
  "format": "dd.MM.yyyy",
  "direction": "horizontal",
  "icon": "https://www.example.com/icon.png"
},
```

### Add title and elements

<br>

#### `textResourceBindings.title`

> The title for the date is shown as a label and can be added as text directly or referenced via a text key to a [text resource](/nb/altinn-studio/reference/ux/texts/#legge-til-og-endre-tekster-i-en-app).

Reviewer: ```suggestion
Label for the date. This can be added as a hardcoded string or as a reference to a [text resource](/nb/altinn-studio/reference/ux/texts/#legge-til-og-endre-tekster-i-en-app).
```

#### `value`

Displays the value for the label and can be added as a ISO-date string or an expression.

#### `format`

Sets the formatting of the date provided in the value field. This is based on [Unicode Technical standard](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table).

#### `direction`

Decides if the date should is shown under or next to the label. horizontal for "next to" and vertical for "under". 

#### `icon`

> If you want to show an icon before the label, you can add an URL path.

Author: Ja godt poeng, vet ikke helt hva som har skjedd her. Mente jeg rettet opp buggen, så har den kommet tilbake igjen. Kanskje droppe å ha det med intil videre?.

{{% /notice %}}

| **Property**                 | **Type** | **Description**                                                                          |
|------------------------------|----------|------------------------------------------------------------------------------------------|
| `id`                         | string   | The component ID. Must be unique within all layouts/pages in a layout-set. Cannot end with <dash><number>. |
| `value`                      | string   |  A hardcoded string or expression for resolving the date you want to display. Must resolve to a ISO6801 string. |
| `textResourceBindings.title` | string   | A label for the date you want to display.                                                   |
| `format`                     | string   | A formatting string based on [Unicode Technical standard](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table).                             |
| `direction`                  | string   | Sets the structured direction of label and value. <br/><br/>**Enum:** [horizontal, vertical] . <br/><br/>**Default:** horizontal. |

> | `icon`                       | string   | A URL path to the image/icon.                                                            |

Reviewer: Fjernes ref kommentar under?

## Configuration

### Add component

```json{hl_lines="6-"}
{
  "id": "date",
  "type": "Date",
  "textResourceBindings": {
    "title": "Date"
  },
  "value": "2022-09-27 18:00:00.000",
  "format": "dd.MM.yyyy",
  "direction": "horizontal",
  "icon": "https://www.example.com/icon.png"
},
```

### Add title and elements

<br>

#### `textResourceBindings.title`

Label for the date. This can be added as a hardcoded string or as a reference to a [text resource](/nb/altinn-studio/reference/ux/texts/#legge-til-og-endre-tekster-i-en-app).

#### `value`

The value of the date to be displayed. Can be added as a string or an expression.

#### `format`

Sets the formatting of the date provided in the value field. This is based on [Unicode Technical standard](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table).

#### `direction`

Decides if the date should be shown under or next to the label. `horizontal` for showing it next to the label and `vertical` for under. 

<!-- 
#### `icon`

If you want to show an icon before the label, you can add a URL path to the icon/image you want to be displayed.
-->

<br>

> #### Example

Reviewer: Fjernes ref kommentar under?
